openedx/features/enterprise_support/api.py

```diff
@@ -233,6 +234,32 @@ def enterprise_enabled():
     return 'enterprise' in settings.INSTALLED_APPS and getattr(settings, 'ENABLE_ENTERPRISE_INTEGRATION', True)
 
 
+def enterprise_customer_for_request(request, tpa_hint=None):
```

Comment:
Ah, I see where you are coming from now with respect to the `enterprise_support` module. Makes sense.

Given that approach, seems like we should copy this function over to edx-enterprise and change the function here so that it calls the function from edx-enterprise. This way if we decide to move to an Enterprise IDA, we just swap out the call to the edx-enterprise function with an API call. Sound good?

I still don't see a reason why the `enterprise-sidebar.html` template shouldn't live in the edx-enterprise repo. I'm not sure that could be considered tech debt since it is something new we are adding with this PR.